### PR TITLE
Add supports for zoltan and metis to build agglomerates

### PR DESCRIPTION
**Pull request by Rombur**
This adds the possibility to use zoltan or metis to create the agglomerate. Since zoltan and metis are called locally on each processor, agglomerates cannot cross processor boundary. This PR adds a new function `build_agglomerates_partitioner` and move `build_agglomerates` to `build_agglomerates_block`.

### CODE REVIEW

File: include/mfmg/amge.templates.hpp [hunk 1]
```diff
@@ -34,89 +37,33 @@ AMGe<dim, VectorType>::AMGe(MPI_Comm comm,
...
-  unsigned int constexpr x_p = 1;
-  unsigned int constexpr y_p = 3;
-  unsigned int constexpr z_p = 5;
-
-  // Flag the cells to create the agglomerates
-  unsigned int agglomerate = 1;
-  for (auto cell : _dof_handler.active_cell_iterators())
+  std::string partitioner_type = ptree.get<std::string>("partitioner");
+  std::transform(partitioner_type.begin(), partitioner_type.end(),
+                 partitioner_type.begin(), ::tolower);
```
Comment (dalg24) [inline include/mfmg/amge.templates.hpp]: `std::tolower` from `<locale>`?

File: include/mfmg/amge.templates.hpp [hunk 2]
```diff
@@ -302,6 +249,171 @@ void AMGe<dim, VectorType>::compute_restriction_sparse_matrix(
...
+        }
+        else
+          break;
+      }
+
+      ++agglomerate;
+    }
+  }
+
+  return agglomerate - 1;
```
Comment (dalg24) [inline include/mfmg/amge.templates.hpp]: Is that because you meant to sent non-zero user indices?

File: include/mfmg/amge.templates.hpp (hunk 1 above)
Comment (Rombur) [reply include/mfmg/amge.templates.hpp]: no it's a C function but it's similar

File: include/mfmg/amge.templates.hpp (hunk 2 above)
Comment (Rombur) [reply include/mfmg/amge.templates.hpp]: Yes, I just wanted an easy way to check that the cells were flagged correctly that's why I didn't want to use zero which is the default value of the user index

**Comment by Rombur** [comment]
How cruel of you to let a PR alone and afraid! Will someone have pity and merge it?

**Comment by Rombur** [comment]
Thanks!

**Pull request merged**